### PR TITLE
docs: First pass fleshing out the user guide

### LINKED ISSUE #695: Update new docs
- [ ] Write a "quick-start" showing how to get started
- [ ] Review / update the "Why Kaskada" (maybe combine with "Tour")?
- [ ] Review the "Tour"
- [ ] Verify installation instructions (https://kaskada.io/kaskada/guide/installation.html)
- [x] Explain "timestreams" (https://kaskada.io/kaskada/guide/timestreams.html)
- [x] Review "Data Types" (https://kaskada.io/kaskada/guide/data_types.html)
- [ ] Review / update entities and grouping (https://kaskada.io/kaskada/guide/entities.html)
- [ ] Write aggregations (include examples / plots) (https://kaskada.io/kaskada/guide/aggregation.html)
- [ ] Write joins (implicit, explicit, etc.) (https://kaskada.io/kaskada/guide/joins.html)
- [ ] "Guide" level introduction to sources (https://kaskada.io/kaskada/guide/sources.html)
- [ ] Docs for time and data windowed aggregations

### PR DESCRIPTION
**Pull request by bjchambers**
This closes #695.

### CODE REVIEW

File: python/docs/source/guide/execution.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Execution
+
+A Timestream may be executed and written to a [destination](#destinations) in a variety of ways.
+The things to consider when deciding how to execute the Timestream are:
+
+1. Whether you want the _history_ of points or the _snapshot_ of values for each entity at a given time.
+2. Whether you want to run a single _query_ or start a _materialization_.
```
Comment (kerinin) [inline python/docs/source/guide/execution.md:7]: Re-word per discussion about "once" vs "live"

File: python/docs/source/guide/execution.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+```{admonition} Preview during development
+:class: tip
+
+While developing queries, it is often useful to view a few rows from the result.
+Using {py:meth}`kaskada.Timestream.preview` you can retrieve a small set of rows from the result set as a Pandas DataFrame.
+```
+
+## History vs. Snapshot
+
+Executing a Timestream for the history outputs every point in the Timestream.
```
Comment (kerinin) [inline python/docs/source/guide/execution.md:22]: Link to `Timestream` explanation, if we have one? Maybe not here, but somewhere in the page.

**Review by kerinin — APPROVED** [review]

File: python/docs/source/guide/execution.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+Similarly, when producing a history, Kaskada can use any persisted state before the earliest new event and before the "changed since" time.
+
+## Destinations
+
+The {py:class}`kaskada.Result` class provides a variety of ways of retrieving the results within Python.
+These include Pandas, PyArrow and Python dictionaries, as well as iterators and async iterators over the same.
+This allows you to run the entire retrieve-evaluate-respond loop within a single Python process.
+
+```{todo}
+Expose and document additional destinations such as Parquet, Kafka, Pulsar, etc.
```
[the remark this answers is not in the document]
Comment (bjchambers) [reply python/docs/source/guide/execution.md:89]: I think it already does -- the `{py:class}'kaskada.Result'` links to the documentation for that class.

File: python/docs/source/quickstart.md [hunk 4]
```diff
@@ -9,10 +9,23 @@ mystnb:
...
+This shows the bare minimum needed to get started with Kaskada.
+
+## Install
+
+Install the latest version.
+This uses `kaskdaa>=0.6.0-a.0` to ensure the pre-release version is installed.
 
-Write the quick start.
 ```
+pip install kaskada>=0.6.0-a.0
```
[the remark this answers is not in the document]
Comment (bjchambers) [reply python/docs/source/quickstart.md:20]: The `>=` means it should pull the latest pre-release version, so yes.

**Pull request merged**